fuzz/smime.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * https://www.openssl.org/source/license.html
+ * or in the file LICENSE in the source distribution.
+ */
+
+#include <stdio.h>
+#include <openssl/err.h>
+#include <openssl/pkcs7.h>
+#include "fuzzer.h"
+
+int FuzzerInitialize(int *argc, char ***argv)
+{
+	return 1;
+}
+
+int FuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    BIO *b = BIO_new_mem_buf(buf, len);
+    PKCS7 *p7 = SMIME_read_PKCS7(b, NULL);
+
+    if (p7) {
+        PKCS7_free(p7);
+    }
```

Comment:
Can you do more things with the PKCS7? For instance call functions like PKCS7_get_signer_info(), PKCS7_cert_from_signer_info(), PKCS7_get_signed_attribute(), PKCS7_get_smimecap(), PKCS7_dataVerify(), ...

Reply:
I guess the CMS fuzzer could use things like that too.

Reply:
Added some usage of the p7 API. I'm candidly not totally confident they're correct, since many of these functions aren't documented.

Reply:
You might want to look at the pkcs7 and smime tool for other things that might be useful to do.